src/pages/surveys/[id]/questions/[order].page.tsx

```diff
@@ -0,0 +1,79 @@
+import type { NextPage } from 'next';
+import { useRouter } from 'next/router';
+import Image from 'next/image';
+import { useContext, useEffect, useState } from 'react';
+
+import Button from 'components/Button';
+import PageLoader from 'components/PageLoader';
+import SurveyQuestion from 'components/SurveyQuestion';
+import useSession from 'hooks/useSession';
+import { SurveyQuestion as SurveyQuestionInterface } from 'services/surveyInterfaces';
+import { StoreContext } from 'store';
+
+import styles from './[order].module.css';
+
+const Question: NextPage = () => {
+  useSession();
+  const router = useRouter();
+  const questionOrder = parseInt(router.query.order as string);
+  const surveyId = router.query.id as string;
+
+  const {
+    store: { currentSurvey },
+  } = useContext(StoreContext);
+
+  const [currentQuestion, setCurrentQuestion] =
+    useState<SurveyQuestionInterface>({} as SurveyQuestionInterface);
+  const [isLoading, setIsLoading] = useState(true);
+  const [answers, setAnswers] = useState(null);
+
+  useEffect(() => {
+    const question = currentSurvey?.questions.find(
+      ({ displayOrder }) => displayOrder === questionOrder
+    );
+
+    if (question !== undefined) {
+      setCurrentQuestion(question);
+      setIsLoading(false);
+    } else {
+      router.push(`/surveys/${surveyId}`);
+    }
+  }, [questionOrder]);
+
+  const totalQuestion = currentSurvey?.questions.length || 0;
```

Comment:
```suggestion
  const questionsCount = currentSurvey?.questions.length || 0;
```
or `totalNumberOfQuestions` or then just `numberOfQuestions`, or `lastQuestionOrder` (to remaind in the "order" thing) ;-) 
(yeah, that was a live-brainstorm... ^^)

Reply:
I just read your test below, as you mentioned `given question order is not the last one`, which makes me more in favor of `lastQuestionOrder` (tests are a good place to challenge variable names too 😎)

Reply:
Totally agree with you 👍 Using the same keyword among implementations and tests improves code readability.